tenant/store.go

```diff
@@ -49,3 +48,149 @@ type Segment struct {
 	Name     string
 	Seq      uint64
 }
+
+func (tenantStore *tenantStore) listTenants() ([]Tenant, error) {
+	var tenants []Tenant
+	log.Println("In listTenants()", &tenants)
+	tenantStore.DbStore.Db.Find(&tenants)
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return nil, err
+	}
+	log.Println(tenants)
+	return tenants, nil
+}
+
+func (tenantStore *tenantStore) listSegments(tenantId uint64) ([]Segment, error) {
+	var segments []Segment
+	log.Println("In listTenants()", &segments)
+	tenantStore.DbStore.Db.Where("tenant_id = ?", tenantId).Find(&segments)
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return nil, err
+	}
+	log.Println(segments)
+	return segments, nil
+}
+
+func (tenantStore *tenantStore) addTenant(tenant *Tenant) error {
+	var tenants []Tenant
+	tenantStore.DbStore.Db.Find(&tenants)
+
+	//	myId := tenant.Id
+	tenant.Seq = uint64(len(tenants) + 1)
+	tenantStore.DbStore.Db.Create(tenant)
+	tenantStore.DbStore.Db.NewRecord(*tenant)
+
+	// TODO better way of getting sequence
+	//
+	//	var tenantSeq uint64
+	//	for i := range tenants {
+	//		if tenants[i].Id == myId {
+	//			tenantSeq = uint64(i)
+	//			tenant.Seq = tenantSeq
+	//			tenantStore.DB().Save(tenant)
+	//			log.Printf("Sequence for tenant %s is %d\n", tenants[i].Name, tenantSeq)
+	//			break
+	//		}
+	//	}
+
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (tenantStore *tenantStore) findTenant(id uint64) (Tenant, error) {
+	var tenants []Tenant
+	log.Println("In findTenant()")
+	tenantStore.DbStore.Db.Find(&tenants)
+	err := common.MakeMultiError(tenantStore.DbStore.Db.GetErrors())
+	if err != nil {
+		return Tenant{}, err
+	}
+	for i := range tenants {
+		if tenants[i].Id == id {
+			return tenants[i], nil
+		}
+	}
+	return Tenant{}, errors.New("Not found")
+	//	tenant := Tenant{}
+	//	tenantStore.DB().Where("id = ?", id).First(&tenant)
+	//	err := common.MakeMultiError(tenantStore.DB().GetErrors())
+	//	if err != nil {
+	//		return tenant, err
+	//	}
+	//	return tenant, nil
+}
+
+func (tenantStore *tenantStore) addSegment(tenantId uint64, segment *Segment) error {
+	var err error
+
+	// TODO better way of getting sequence
+	var segments []Segment
+	tenantStore.DbStore.Db.Where("tenant_id = ?", tenantId).Find(&segments)
+	segment.Seq = uint64(len(segments) + 1)
```

Comment:
ID is not the same as sequence. ID is DB's autogenerated ID, can be used as a resource ID in rest requests, etc, but is not helpful for keeping track of actual sequential number of existing entities.